airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/buffered_stream_consumer/BufferedStreamConsumer.java

```diff
@@ -155,29 +168,42 @@ protected void acceptTracked(final AirbyteMessage message) throws Exception {
            */
         }
       }
-      /*
-       * TODO: (ryankfu) after record has added and time has been met then to see if time has elapsed then
-       * flush the buffer
-       *
-       * The reason that this is where time component should exist here is primarily due to #acceptTracked
-       * is the method that processes each AirbyteMessage. Method to call is
-       * bufferingStrategy.flushWriter(stream, streamBuffer)
-       */
     } else if (message.getType() == Type.STATE) {
       stateManager.addState(message);
     } else {
       LOGGER.warn("Unexpected message: " + message.getType());
     }
+    periodicBufferFlush();
   }
 
   /**
-   * After marking states as committed, emit the state message to platform then clear state messages
-   * to avoid resending the same state message to the platform
+   * After marking states as committed, return the state message to platform then clear state messages
+   * to avoid resending the same state message to the platform. Also updates the last flushed time
+   * since it can be deterministic that a state message can be returned to the platform
    */
   private void markStatesAsFlushedToDestination() {
     stateManager.markPendingAsCommitted();
     stateManager.listCommitted().forEach(outputRecordCollector);
     stateManager.clearCommitted();
+    lastFlushEmission = Instant.now();
+  }
+
+  /**
+   * Periodically flushes buffered data to destination storage when exceeding BUFFER_FLUSH_FREQUENCY.
+   * Also resets the last time a flush occurred
+   */
+  private void periodicBufferFlush() {
+    final Duration timeElapsed = Duration.between(lastFlushEmission, Instant.now());
+    // When the last time the buffered has been flushed exceed the frequency, flush the current
+    // buffer before receiving incoming AirbyteMessage
+    if (timeElapsed.compareTo(BUFFER_FLUSH_FREQUENCY) >= 0) {
```

Comment:
minor readability improvement
```suggestion
    if (Instant.now().isAfter(lastFlushEmission.plus(BUFFER_FLUSH_FREQUENCY))) {
```

also, we could just store `nextFlushDeadline = Instant.now().plus(BUFFER_FLUSH_FREQUENCY)` in markStatesAsFlushedToDestination rather than needing to recalculate `lastFlushEmission.plus(...)` for every new message